Newsletters: add 249 (2023-05-03) (#1099)
bitcoinops/bitcoinops.github.io: _posts/en/newsletters/2023-05-03-newsletter.md

    In this week's post, Ingala sought to demonstrate that MATT would not
    only be very flexible, but that it would also be efficient and easy
    to use in transaction templates that might one day see frequent use.
    As is done in recent versions of the `OP_VAULT` proposal, Ingala builds
    upon the [BIP119][] proposal for [OP_CHECKTEMPLATEVERIFY][topic
    op_checktemplateverify] (CTV).  Using two additional proposed
    opcodes (and acknowledging that they don't entirely cover everything
    necessary), he provides a set of features that is almost equivalent
    to `OP_VAULT`.  It's only notable omission is "an option to add an
    additional output that is sent back to the same exact vault."

    As of this writing, Ingala's post had not received any direct
    replies, but there was [continued discussion][halseth matt] about
    his original proposal for MATT and it's ability to allow
    verification that an (essentially) arbitrarily complex program was
    run.

- **Analysis of signature adaptor security:** Adam Gibson
  [posted][gibson adaptors] to the Bitcoin-Dev mailing list an analysis
  of the security of [signature adaptors][topic adaptor signatures],

>   especially about how they interact with [multisignatures][topic

Reviewer: Maybe

```suggestion
  especially about how they interact with [multisignature][topic
```

[Bitcoin Inquisition][bitcoin inquisition repo].*

- [Bitcoin Core #25158][] adds an `abandoned` field to the transaction detail
  responses from the `gettransaction`, `listtransactions`, and `listsinceblock` RPCs
  indicating which transactions have been marked [abandoned][abandontransaction rpc].

- [Bitcoin Core #26933][] reintroduces the requirement that each
  transaction meet the node's minimum relay feerate (`-minrelaytxfee`)
  in order to be accepted to the mempool, even when being evaluated as a
  package. Package validation still allows bumping a transaction below
  the dynamic mempool minimum feerate. This policy was reintroduced to
  avoid the risk of zero-fee transactions losing their fee-bumping
  descendant in the event of a replacement. It may reversed in the
  future if a DoS-resistant method of preventing such transactions is
  found, e.g. through a package topology restriction like v3 or a
  modification to the mempool's eviction process.

- [Bitcoin Core #25325][] introduces a pool based memory resource for
  the UTXO cache. The new data structure pre-allocates and manages a
  larger pool of memory to track UTXOs instead of allocating and freeing

>   memory for each UTXO individually. UTXO lookups cause a major part of

Reviewer: Maybe

```suggestion
  memory for each UTXO individually. UTXO lookups represent a major proportion of
```

  in a PSBT needs to include a full copy of the transaction which
  contained that output so that a signer can verify the amount of that
  output.  Creating a spend without verifying the amount of the output
  being spent can lead to the spender massively overpaying transaction
  fees, so verification is important.

  Each spend of a segwit v0 output commits to its amount in an attempt
  to allow PSBTs to include only the output's scriptPubKey and amount
  rather than the entire previous transaction.  It was believed that
  this would eliminate the need to include the entire transaction.
  Since every unspent transaction output for every confirmed transaction
  is stored in Bitcoin Core's UTXO set, the `utxoupdatepsbt` RPC can add
  the necessary scriptPubKey and amount data to any PSBT spending a
  UTXO.  The `utxoupdatepsbt` also previously searched the local node's
  mempool for UTXOs to allow users to spend outputs from unconfirmed
  transactions.

  However, after `utxoupdatepsbt` was added to Bitcoin Core, several
  hardware signing devices began requiring even spends of segwit v0
  outputs to include full transactions in order to prevent a variation

>   on fee overpayment that could result from a user seemingly signing the

Reviewer: ```suggestion
  of fee overpayment that could result from a user seemingly signing the
```
